manager/integration/tests/test_ha.py

```diff
@@ -2370,4 +2359,68 @@ def test_replica_failure_during_attaching():
     12. Wait for volume2 becoming Healthy.
     13. Verify data content and r/w capability for volume2.
     """
-    pass
+
+    replenish_wait_setting = \
+        client.by_id_setting(SETTING_REPLICA_REPLENISHMENT_WAIT_INTERVAL)
+    client.update(replenish_wait_setting, value='10')
+
+    replica_node_soft_anti_affinity_setting = \
+        client.by_id_setting(SETTING_REPLICA_NODE_SOFT_ANTI_AFFINITY)
+    client.update(replica_node_soft_anti_affinity_setting, value="false")
+
+    volume_name_1 = volume_name
+    host_id = get_self_host_id()
+    node = client.by_id_node(host_id)
+
+    extra_disk_path = create_host_disk(client, volume_name_1, str(3 * Gi),
+                                       host_id)
+    extra_disk = {"path": extra_disk_path, "allowScheduling": True}
+
+    update_disks = get_update_disks(node.disks)
+    default_disk_name = next(iter(update_disks))
+    update_disks["extra-disk"] = extra_disk
+    update_disks[default_disk_name].allowScheduling = False
+
+    node = node.diskUpdate(disks=update_disks)
+    node = common.wait_for_disk_update(client, node.name, len(update_disks))
+
+    volume_name_2 = volume_name + '-2'
+    volume_2 = create_and_check_volume(client, volume_name_2, 3, str(1 * Gi))
+    volume_2.attach(hostId=host_id)
+    volume_2 = common.wait_for_volume_healthy(client, volume_name_2)
+    write_volume_random_data(volume_2)
+    volume_2.detach()
+    wait_for_volume_detached(client, volume_name_2)
+
+    # unmount the disk
+    mount_path = os.path.join(DIRECTORY_PATH, volume_name_1)
+    common.umount_disk(mount_path)
+    cmd = ['rm', '-r', mount_path]
+    subprocess.check_call(cmd)
+
+    volume_2 = client.by_id_volume(volume_name_2)
+    volume_2.attach(hostId=host_id)
+    common.wait_for_volume_status(client, volume_name_2,
+                                  VOLUME_FIELD_ROBUSTNESS,
+                                  VOLUME_ROBUSTNESS_DEGRADED)
+
+    update_disks[default_disk_name].allowScheduling = True
+    update_disks["extra-disk"]["allowScheduling"] = False
+
+    node = node.diskUpdate(disks=update_disks)
+    common.wait_for_disk_update(client, node.name, len(update_disks))
+
+    common.wait_for_volume_status(client, volume_name_2,
+                                  VOLUME_FIELD_ROBUSTNESS,
+                                  VOLUME_ROBUSTNESS_HEALTHY)
+
+    volume_2 = client.by_id_volume(volume_name_2)
+    write_volume_random_data(volume_2)
+
+    del update_disks["extra-disk"]
+    node.diskUpdate(disks=update_disks)
+    common.wait_for_disk_update(client, node.name, 1)
+
+    replenish_wait_setting = \
+        client.by_id_setting(SETTING_REPLICA_REPLENISHMENT_WAIT_INTERVAL)
+    client.update(replenish_wait_setting, value='600')
```

Comment:
nodes and settings will be reset automatically

Reply:
fixed